lib/common/services/storageserviceclient.js

```diff
@@ -478,7 +478,7 @@ StorageServiceClient.prototype._performRequest = function (webResource, body, op
                 finalRequestOptions.headers['content-length'] :
                 Constants.BlobConstants.MAX_SINGLE_UPLOAD_BLOB_SIZE_IN_BYTES;
 
-              var concatBuf = new Buffer(size);
+              var concatBuf = Buffer.alloc(parseInt(size));
```

Comment:
this line seemed to have a bug in it, that was caught after I converted to `alloc`. The content-length header that is conditionally used for `size`  may be being returned as a string, when you might have been expecting a number?